lecture_1/lecture_1_passed/passed.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"time"
+)
+type passedAPIResponse struct{
+	Name string
+	Age int
+	Passed bool
+	Skills []string
+}
+
+
+
+const passedUrl = "https://run.mocky.io/v3/f7ceece5-47ee-4955-b974-438982267dc8"
+
+func linearBackoff(retry int) time.Duration {
+	return time.Duration(retry) * time.Second
+}
+
+func writingToFile(f *os.File, decodedContent []passedAPIResponse){
+	for i := range decodedContent{
+		if decodedContent[i].Passed{
+			var found = false
+			//var knownSkills []string
+			var knownSkillText = strings.Join(decodedContent[i].Skills, ", ")
+			for _,val := range decodedContent[i].Skills{
+				//knownSkills = append(knownSkills, val)
+
+				if val == "Java" || val == "Go"{
+					found = true
+				}
+			}
+			if found{
+				defer f.WriteString(fmt.Sprintf("%s - %v\n", decodedContent[i].Name, knownSkillText))
```

Comment:
Why the `defer`?

Reply:
Yea, that was an oversight, looking back it is a big mistake since it reverses the order in the output. I have since fixed the problem. Thank you for the review.  New commit is here: https://github.com/Rzucky/code-cadets-2021/pull/1/commits/bc9d70a17a549a0f43995c32cf9d0ea53e8d75e0
and also in Google Docs since Slack isn't working